### DISCUSSION TITLE
Plotting Delaunay triangulations and associated surfaces

### DISCUSSION THREAD

**Discussion by jreniel** [plotters-rs/plotters#542] [Q&A]
Hello! 
I have some numerical analysis code in Rust, but I would like to have some visualization capabilities.
In particular, I am trying to understand how I would go about to plotting Delaunay Triangulations along with the associated surfaces. I would also like to open a window with the plot during runtime, like matplotlib does, but I am not sure if this is possible without using GTK. As of now, plotting has been a big pain point for me with Rust, and I'm loving Rust, and want to continue using it, but I've reached the point in my development where plotting has become super essential, and I'm figuratively begging for help to get some pointers. I'm sure that if I get the gist of this, I will be able to build lots of cool stuff!!
Thanks for your help!

**Comment by AaronErhardt** [comment]
Actually, egui has some really nice plotting capabilities, especially if you need something interactive (and not too complex). On the other side, plotters works with most GUI frameworks in Rust. 
I have to agree with you though that plotting in Rust is not ideal in general. Plotters has a lot of technical debt which makes some parts of the API very unpleasant to use. And since the original developer discontinued his efforts, only a few people like me occasionally merge minor improvements, but the major issues still persist.

**Reply by jreniel** [reply]
Thank for your reply @AaronErhardt. It is important for me to be aware of these details so I can make an informed choice for the plotting on my crates. In any case, if possible, could you please clarify how I would go about to plotting tessellations and Voronoi complexes using plotters? Regardless of which crate I chose for the plotting in the end, it would still be nice to know! 
Thanks!

**Reply by AaronErhardt** [reply]
> In any case, if possible, could you please clarify how I would go about to plotting tessellations and Voronoi complexes using plotters?

I guess you have to use something like polygons and paths and calculate the coordinates of the points yourself: https://docs.rs/plotters/latest/plotters/element/index.html#structs
AFAIK there's no crate that supports plotting those specific things out of the box, but you can do your own research. I'm not familiar with plotting such things, so I can only make assumptions.

EDIT: sorry if you got a notification twice, I had bad internet and it was shown as not sent despite being sent.
